src/main/java/forestry/factory/tiles/TileMillRainmaker.java

```diff
@@ -103,15 +114,22 @@ public void activate() {
         }
 
         if (!worldObj.isRemote) {
-            if (reverse) {
-                worldObj.getWorldInfo().setRaining(false);
-            } else {
-                worldObj.getWorldInfo().setRaining(true);
-                worldObj.getWorldInfo().setRainTime(duration);
+            switch (weather) {
+                case CLEAR:
+                    worldObj.getWorldInfo().setRaining(false);
+                    worldObj.getWorldInfo().setThundering(false);
+                    break;
+                case STORM:
+                    worldObj.getWorldInfo().setRaining(true);
+                    worldObj.getWorldInfo().setThundering(true);
+                    break;
+                case RAIN:
+                    worldObj.getWorldInfo().setRaining(true);
+                    worldObj.getWorldInfo().setThundering(false);
+                    break;
```

Comment:
Missing setting the weather time
```suggestion
                case STORM:
                    worldObj.getWorldInfo().setRaining(true);
                    worldObj.getWorldInfo().setRainTime(duration);
                    worldObj.getWorldInfo().setThundering(true);
                    worldObj.getWorldInfo().setThunderTime(duration);
                    break;
                case RAIN:
                    worldObj.getWorldInfo().setRaining(true);
                    worldObj.getWorldInfo().setRainTime(duration);
                    worldObj.getWorldInfo().setThundering(false);
                    break;
```